### PR TITLE
build: add gen-csv files in downstream repo

### PR DESCRIPTION
**Pull request by subhamkrai**
build: add gen-csv files in downstream repo

This commit add the csv related changes to downstream and
later will be removed from upstream also renaming make 'csv'
to 'gen-csv' and also adds validation check in ci to verify
right files are modified.

build: add csv envs to rook deploy envs

moving all the default envs which were part of
ocs-op rook csv to operator deployment.

build: these are auto-generated files

This commit contains auto-generated files
that are added/modified by running 'make gen-csv'.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by subhamkrai** [comment]
@travisn @Nikhil-Ladha PTAL

**Comment by Nikhil-Ladha** [comment]
cc @iamniting

**Comment by BlaineEXE** [comment]
I am also somewhat concerned that this code is special for downstream and that it may be erased when github.com/rook/rook@master is resynced here, or to release-\* branches. I certainly may also be misunderstanding something here.

**Comment by subhamkrai** [comment]
> I am also somewhat concerned that this code is special for downstream and that it may be erased when github.com/rook/rook@master is resynced here, or to release-* branches. I certainly may also be misunderstanding something here.

If we just skip one upstream commit that removes the csv related file, downstream should be good then I guess

**Comment by Nikhil-Ladha** [comment]
> > I am also somewhat concerned that this code is special for downstream and that it may be erased when github.com/rook/rook@master is resynced here, or to release-* branches. I certainly may also be misunderstanding something here.
> 
> If we just skip one upstream commit that removes the csv related file, downstream should be good then I guess

We don't have to skip anything as such, once we are set to merge these changes at that time we should first delete the changes in upstream, sync those changes to downstream, rebase this PR and then merge. Then, everything should be good and we won't be facing any challenges.

And, later syncs won't be a problem because we are just bring commits from upstream to downstream, that would continue as usual.

**Comment by subhamkrai** [comment]
> > > I am also somewhat concerned that this code is special for downstream and that it may be erased when github.com/rook/rook@master is resynced here, or to release-* branches. I certainly may also be misunderstanding something here.
> > 
> > 
> > If we just skip one upstream commit that removes the csv related file, downstream should be good then I guess
> 
> We don't have to skip anything as such, once we are set to merge these changes at that time we should first delete the changes in upstream, sync those changes to downstream, rebase this PR and then merge. Then, everything should be good and we won't be facing any challenges.
> 
> And, later syncs won't be a problem because we are just bring commits from upstream to downstream, that would continue as usual.

IIRC, once we remove the csv generation files from upstream and sync that to downstream those file will be removed in downstream also and the csv generation will fail. But let's sync on this @Nikhil-Ladha

**Comment by parth-gr** [comment]
@subhamkrai in last offline discussion we suggested to have a separate readme.md that only talks about downstream specific changes and avoid merge conflicts.
Currently this change and change for SVG is added https://github.com/red-hat-storage/rook/pull/589
So with this change have a plan to initiate that

cc @travisn @iamniting

**Comment by iamniting** [comment]
> @subhamkrai in last offline discussion we suggested to have a separate readme.md that only talks about downstream specific changes and avoid merge conflicts. Currently this change and change for SVG is added #589 So with this PR have a plan to initiate that
> 
> cc @travisn @iamniting

I would highly suggest doing so, so that we don't miss any commits in the future branches.

**Comment by subhamkrai** [comment]
> > @subhamkrai in last offline discussion we suggested to have a separate readme.md that only talks about downstream specific changes and avoid merge conflicts. Currently this change and change for SVG is added #589 So with this PR have a plan to initiate that
> > cc @travisn @iamniting
> 
> I would highly suggest doing so, so that we don't miss any commits in the future branches.

I will prefer to do it in separate PR
since I'm planning to get this PR merged EOD and docs PRs generally have more reviews before they get merged

**Comment by parth-gr** [comment]
> I will prefer to do it in separate PR since I'm planning to get this PR merged EOD and docs PRs generally have more reviews before they get merged

I don't have any problem with separate PR, but keep that item in mind as we cant track it separately anywhere.

File: build/csv/rook-ceph.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,3285 @@
...
+  version: {{.RookOperatorCsvVersion}}
+  replaces: rook-ceph.v1.1.1
+  minKubeVersion: 1.10.0
+  labels:
+    alm-owner-etcd: rookoperator
+    operated-by: rookoperator
+  selector:
+    matchLabels:
+      alm-owner-etcd: rookoperator
+      operated-by: rookoperator
```
Comment (Nikhil-Ladha) [inline build/csv/rook-ceph.clusterserviceversion.yaml:3278]: Update these fields as per ocs-operator csv

File: build/csv/rook-ceph.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,3285 @@
...
+  maintainers:
+    - name: Red Hat, Inc.
+      email: customerservice@redhat.com
+  maturity: alpha
+  provider:
+    name: Provider Name
+    url: https://your.domain
+  version: {{.RookOperatorCsvVersion}}
+  replaces: rook-ceph.v1.1.1
+  minKubeVersion: 1.10.0
```
Comment (Nikhil-Ladha) [inline build/csv/rook-ceph.clusterserviceversion.yaml:3271]: Please update these fields as per the ocs-operator CSV

**Comment by subhamkrai** [comment]
@iamniting PTAL

File: build/csv/rook-ceph.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,3281 @@
...
+    - shared filesystem
+  links:
+    - name: Source Code
+      url: https://github.com/red-hat-storage/rook
+  maintainers:
+    - name: Red Hat Support
+      email: ocs-support@redhat.com
+  maturity: alpha
+  provider:
+    name: Provider Name
```
Comment (iamniting) [inline build/csv/rook-ceph.clusterserviceversion.yaml:3278]: Can you update this as well?

**Review by iamniting — CHANGES_REQUESTED** [review]

File: deploy/examples/operator-openshift.yaml [hunk 4]
```diff
@@ -678,76 +678,111 @@ spec:
...
             - name: ROOK_CURRENT_NAMESPACE_ONLY
+              valueFrom:
+                configMapKeyRef:
+                  key: ROOK_CURRENT_NAMESPACE_ONLY
+                  name: ocs-operator-config
+            - name: CSI_REMOVE_HOLDER_PODS
+              valueFrom:
+                configMapKeyRef:
+                  key: CSI_REMOVE_HOLDER_PODS
+                  name: ocs-operator-config
```
Comment (subhamkrai) [inline deploy/examples/operator-openshift.yaml:689]: @iamniting added the field

Comment (iamniting) [reply deploy/examples/operator-openshift.yaml:689]: looks good

**Comment by iamniting** [comment]
CSV changes looks good to me

**Pull request merged**